eva/udfs/object_detector.py

```diff
@@ -0,0 +1,238 @@
+            "orange",
+            "broccoli",
+            "carrot",
+            "hot dog",
+            "pizza",
+            "donut",
+            "cake",
+            "chair",
+            "couch",
+            "potted plant",
+            "bed",
+            "N/A",
+            "dining table",
+            "N/A",
+            "N/A",
+            "toilet",
+            "N/A",
+            "tv",
+            "laptop",
+            "mouse",
+            "remote",
+            "keyboard",
+            "cell phone",
+            "microwave",
+            "oven",
+            "toaster",
+            "sink",
+            "refrigerator",
+            "N/A",
+            "book",
+            "clock",
+            "vase",
+            "scissors",
+            "teddy bear",
+            "hair drier",
+            "toothbrush",
+        ]
+
+    def forward(self, frames: Tensor) -> pd.DataFrame:
+        """
+        Performs predictions on input frames
+        Arguments:
+            frames (np.ndarray): Frames on which predictions need
+            to be performed
+
+        Returns:
+            tuple containing predicted_classes (List[List[str]]),
+            predicted_boxes (List[List[BoundingBox]]),
+            predicted_scores (List[List[float]])
+
+        """
+        predictions = self.model(frames)
+        outcome = pd.DataFrame()
+        for prediction in predictions:
+            pred_class = [
+                str(self.labels[i]) for i in list(self.as_numpy(prediction["labels"]))
+            ]
+            pred_boxes = [
+                [i[0], i[1], i[2], i[3]]
+                for i in list(self.as_numpy(prediction["boxes"]))
+            ]
+            pred_score = list(self.as_numpy(prediction["scores"]))
+            valid_pred = [pred_score.index(x) for x in pred_score if x > self.threshold]
+
+            if valid_pred:
+                pred_t = valid_pred[-1]
+            else:
+                pred_t = -1
+
+            pred_boxes = np.array(pred_boxes[: pred_t + 1])
+            pred_class = np.array(pred_class[: pred_t + 1])
+            pred_score = np.array(pred_score[: pred_t + 1])
+            outcome = outcome.append(
+                {"labels": pred_class, "scores": pred_score, "bboxes": pred_boxes},
+                ignore_index=True,
+            )
+        return outcome
+
+class Resnet50ObjectDetector(ObjectDetector):
+    def __init__(self):
+        super().__init__("fasterrcnn_resnet50_fpn_v2")
+
+class MobilenetObjectDetector(ObjectDetector):
+    def __init__(self):
+        super().__init__("fasterrcnn_mobilenet_v3_large_fpn")
+
+class SsdLiteObjectDetector(ObjectDetector):
+    def __init__(self):
+        super().__init__("ssdlite320_mobilenet_v3_large")
+
+class SsdVggObjectDetector(ObjectDetector):
+    def __init__(self):
+        super().__init__("ssd300_vgg16")
+
+class KeypointObjectDetector(ObjectDetector):
+    def __init__(self):
+        super().__init__("keypointrcnn_resnet50_fpn")
```

Comment:
Create a child class of `ObjectDetector` and pass the model of interest as a string.